From 6725b4f6171bc491a435f4302af2c5c59b82f700 Mon Sep 17 00:00:00 2001
From: Elly <ellyjones@chromium.org>
Date: Thu, 2 Nov 2023 22:46:35 +0000
Subject: [PATCH] Backport "atspi: mark possibly unused gn variables"

This fixes an error in the do_configure task that occurs if the atspi
version is >= 2.49.90. The patch will be included in upstream's 121
release.

Upstream-Status: Backport [https://crrev.com/c/5001687]
Signed-off-by: Max Ihlenfeldt <max@igalia.com>
---
 build/config/linux/atspi2/BUILD.gn | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/build/config/linux/atspi2/BUILD.gn b/build/config/linux/atspi2/BUILD.gn
index d162920..30bc77f 100644
--- a/build/config/linux/atspi2/BUILD.gn
+++ b/build/config/linux/atspi2/BUILD.gn
@@ -21,10 +21,21 @@ if (use_atk) {
     minor = atspi_version[1]
     micro = atspi_version[2]

+    # These aren't necessarily used if atspi is not old enough to require them.
+    # Also, gn considers variables unused if the only use of them is
+    # short-circuited away, so for example if major == 2 and minor == 48, micro
+    # would be unused.
+    not_needed([
+                 "major",
+                 "minor",
+                 "micro",
+               ])
+
     # ATSPI 2.49.90 now defines these for us and it's an error for us to
     # redefine them on the compiler command line.
     # See ATSPI 927344a34cd5bf81fc64da4968241735ecb4f03b
-    if (minor < 49 || (minor == 49 && micro < 90)) {
+    if (major < 2 || (major == 2 && minor < 49) ||
+        (major == 2 && minor == 49 && micro < 90)) {
       defines = [
         "ATSPI_MAJOR_VERSION=$major",
         "ATSPI_MINOR_VERSION=$minor",
